nilearn/_utils/ndimage.py

```diff
@@ -47,3 +47,87 @@ def get_border_data(data, border_size):
         data[:, :, :border_size].ravel(),
         data[:, :, -border_size:].ravel(),
     ])
+
+
+def _peak_local_max(image, min_distance=10, threshold_abs=0, threshold_rel=0.1,
+                    indices=True, num_peaks=np.inf):
+    """Find peaks in an image, and return them as coordinates or a boolean array.
+
+    Peaks are the local maxima in a region of `2 * min_distance + 1`
+    (i.e. peaks are separated by at least `min_distance`).
+
+    NOTE: If peaks are flat (i.e. multiple adjacent pixels have identical
+    intensities), the coordinates of all such pixels are returned.
+
+    Parameters
+    ----------
+    image : ndarray of floats
+        Input image.
+    min_distance : int
+        Minimum number of pixels separating peaks in a region of `2 *
+        min_distance + 1` (i.e. peaks are separated by at least
+        `min_distance`). To find the maximum number of peaks, use `min_distance=1`.
+    threshold_abs : float
+        Minimum intensity of peaks.
+    threshold_rel : float
+        Minimum intensity of peaks calculated as `max(image) * threshold_rel`.
+    indices : bool
+        If True, the output will be an array representing peak coordinates.
+        If False, the output will be a boolean array shaped as `image.shape`
+        with peaks present at True elements.
+    num_peaks : int
+        Maximum number of peaks. When the number of peaks exceeds `num_peaks`,
+        return `num_peaks` peaks based on highest peak intensity.
+
+    Returns
+    -------
+    output : ndarray or ndarray of bools
+
+        * If `indices = True`  : (row, column ...) coordinates of peaks.
+        * If `indices = False` : Boolean array shaped like `image`, with peaks
+          represented by True values.
+
+    Notes
+    -----
+    The peak local maximum function returns the coordinates of local peaks
+    (maxima) in a image. A maximum filter is used for finding local maxima.
+    This operation dilates the original image. After comparison between
+    dilated and original image, peak_local_max function returns the
+    coordinates of peaks where dilated image = original.
+
+    This code is mostly adapted from scikit image 0.11.3 release.
```

Comment:
To make life easier for our future selves, you may want to indicate which file you took it or the fully qualified function name e.g. skimage.feature.peak.peak_local_max in this case.

You can probably do something similar each function/file you took from scikit-image.